python/cuda_parallel/cuda/parallel/experimental/iterators/_iterators.py

```diff
@@ -288,6 +294,45 @@ def dereference(state):
         return state[0]
 
 
+class ReverseIteratorKind(IteratorKind):
+    pass
+
+
+def make_reverse_iterator(it):
```

Comment:
Can we add type annotations here? It would be helpful to know that `it` can be ` DeviceArray` here.